pkg/logging/logging.go

```diff
@@ -133,6 +137,33 @@ func LoadLogConfig(dataStore, ns, id string) (LogConfig, error) {
 	return logConfig, nil
 }
 
+func loggingProcessAdapter(driver Driver, config *logging.Config) error {
+	var wg sync.WaitGroup
+	wg.Add(3)
+	stdout := make(chan string, 10000)
+	stderr := make(chan string, 10000)
+	processLogFunc := func(reader io.Reader, dataChan chan string) {
+		defer wg.Done()
+		defer close(dataChan)
+		scanner := bufio.NewScanner(reader)
+		for scanner.Scan() {
+			if scanner.Err() != nil {
+				return
+			}
+			dataChan <- scanner.Text()
+		}
+	}
+
+	go processLogFunc(config.Stdout, stdout)
+	go processLogFunc(config.Stderr, stderr)
+	go func() {
+		defer wg.Done()
+		driver.Process(stdout, stderr)
```

Comment:
should catch the error returns.

Reply:
And maybe we don't need  WaitGroup here as `driver.Process` is a syncable function that will process all data reader from channels and block until channels are closed.

Reply:
No, we need WaitGroup here when we have multiple write target at the same time